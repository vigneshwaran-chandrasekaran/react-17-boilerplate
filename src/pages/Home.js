import React from "react";

function Home(props) {
  return <div>Home</div>;
}

Home.propTypes = {};

export default Home;
