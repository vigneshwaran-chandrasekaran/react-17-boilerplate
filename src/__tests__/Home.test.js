import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';
import Home from 'pages/Home';

afterEach(cleanup);

// Snapshots allows you to test if your component renders correctly so in your case

// https://jestjs.io/docs/en/snapshot-testing

/**
 * A typical snapshot test case renders a UI component, takes a snapshot,
 * then compares it to a reference snapshot file stored alongside the test.
 */

describe('Home page', () => {
	test('renders Home page component', () => {
		render(<Home />);

		screen.debug();
		expect(screen.getByText('Home page')).toBeInTheDocument();
		expect(screen.getByText(/page/)).toBeInTheDocument();
		// expect(screen.getByRole('button')).toBeInTheDocument();
	});

	it('should take a snapshot', () => {
		const { asFragment } = render(<Home />);
		// console.log('asFragment 123', asFragment.toString());
		expect(asFragment(<Home />)).toMatchSnapshot();
	});
});
