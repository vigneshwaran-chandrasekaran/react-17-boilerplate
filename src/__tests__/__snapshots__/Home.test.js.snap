// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    Home page
  </div>
</DocumentFragment>
`;

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    Home page
  </div>
</DocumentFragment>
`;
